contrib/kubernetes/datahub/charts/datahub-gms/templates/deployment.yaml

```diff
@@ -37,70 +37,37 @@ spec:
               protocol: TCP
           env:
             - name: EBEAN_DATASOURCE_USERNAME
-              valueFrom:
-                secretKeyRef:
-                  name: {{ include "datahub-gms.fullname" . }}-secret
-                  key: ebean.datasource.username
+              value: "{{ .Values.global.sql.datasource.username }}"
             - name: EBEAN_DATASOURCE_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  name: {{ include "datahub-gms.fullname" . }}-secret
-                  key: ebean.datasource.password
+                  name: "{{ .Values.global.sql.datasource.password.secretRef }}"
```

Comment:
Another approach that I've seen others embrace is this: https://github.com/helm/helm/issues/2780#issuecomment-403307198 . It also allows dynamic environment variables which is good because Spring Kafka allows for dynamic properties via `spring.kafka.properties.*` see https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-application-properties.html#integration-properties

Reply:
Thanks @jsotelo 
I agree with this approach for definingdynamic properties. The properties and secrets defined in values.yaml till now are mandatory arguments to the chart.
IMO, values.yaml is like an input API to the chart, mix of mandatory and optional parameters. mandatory args should be explicitly mentioned and optional ones can be defined using the dynamic properties(kerberos/sasl configs) like extraEnv, extraVolumes, extraSecrets (https://hackernoon.com/the-art-of-the-helm-chart-patterns-from-the-official-kubernetes-charts-8a7cafa86d12)
Needless to say, mentioning them explicitly also improves the documentation of the charts.

Reply:
I agree, explicit environment variables are the way to go b/c they are self documenting. I was suggesting that in addition to them, we add a way to inject dynamic ones to allow user to configure `spring.kafka.properties` fully. Otherwise I am not sure how we would be able to configure things like `SPRING_KAFKA_PROPERTIES_SASL_KERBEROS_SERVICE_NAME` and `SPRING_KAFKA_PROPERTIES_SECURITY_PROTOCOL`, etc. Trying to keep in-sync with kafka source code and specify all possibilities is not practical.

Reply:
Agree. Thanks for pointing this out and I have added a commit(e395b4b) to address dynamic properties by adding three variables, extraEnvs, extraVolumes and extraVolumeMounts to the values.yaml for each chart.
We are using the same for providing kerberos related properties like  SPRING_KAFKA_PROPERTIES_SECURITY_PROTOCOL, mounting keytab file and kerberos configuration file(krb5.conf) .
Sorry for my miss in cherrypicking it to PR branch earlier.